packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  let userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    userConfig = await userConfig();
+  }
+
+  if (typeof userConfig === 'object' && userConfig !== null && !Array.isArray(userConfig)) {
+    return userConfig;
+  }
+
+  throw new MetadataValidationError(
+    `Metadata store config returned an unkown value, should be an object: "${userConfig}"`
+  );
+}
+
+/**
+ * Get the static configuration file path, based on the metadata context.
+ * This uses any custom name provided, but swaps out the extension for `.json`.
+ */
+export function getStaticConfigFile({
+  projectDir,
+  metadataPath,
+}: Pick<MetadataContext, 'projectDir' | 'metadataPath'>): string {
+  const configFile = path.join(projectDir, metadataPath);
+  const configExtension = path.extname(configFile);
+
+  return path.join(projectDir, `${path.basename(configFile, configExtension)}.json`);
+}
+
+/**
+ * Load the store configuration from a metadata context.
+ * This can load `.json` and `.js` config files, using `require`.
+ * It throws MetadataValidationErrors when the file doesn't exist, or contains errors.
+ * The user is prompted to try anyway when errors are found.
+ */
+export async function loadConfigAsync({
+  projectDir,
+  metadataPath,
+}: Pick<MetadataContext, 'projectDir' | 'metadataPath'>): Promise<MetadataConfig> {
+  const configFile = path.join(projectDir, metadataPath);
+  if (!(await fs.pathExists(configFile))) {
+    throw new MetadataValidationError(`Metadata store config file not found: "${configFile}"`);
+  }
+
+  const configData = await resolveDynamicConfigAsync(configFile);
+  const { valid, errors: validationErrors } = validateConfig(configData);
+
+  if (!valid) {
+    const error = new MetadataValidationError(
+      `Metadata store config errors found`,
+      validationErrors
+    );
+
+    logMetadataValidationError(error);
+    Log.newLine();
+    Log.warn(
+      'Without further updates, the current store configuration may fail to be synchronized with the App Store or pass App Store review.'
```

Comment:
@brentvatne taught me this :D 

```suggestion
      'Without further updates, the current store configuration can fail to be synchronized with the App Store or pass App Store review.'
```

<a href="https://imgflip.com/i/42bw1b"><img src="https://i.imgflip.com/42bw1b.jpg" title="made at imgflip.com"/></a>

Reply:
🤣